### ISSUE TITLE
Stateful rule with operator 'OR' not triggered when only one condition is true

### ISSUE THREAD

**Issue by akreienbring** [crafts69guy/rule-engine-js#10]
With this object:

```
const dog= {
  state: "sleeping",
  hunger: 0,
};
```
and this rule:
```
const rule = {
  or: [
    {
      changedTo: ["state", "awake"],
      increased: ["hunger"],
    },
  ],
};
```
and this code:
```
//first evaluation
statefulEngine.evaluate("hunger", rule, dog);

dog.state = "awake";
dog.hunger = 1;

 statefulEngine.evaluate("hunger", rule, dog).then((result) => {
   console.log(JSON.stringify(result));
 });
```

I get a successfull result:
```
{"success":true,"triggered":true,"hasChangeOperator":true,"stateChange":"triggered"}
```

BUT:  If I only change one property, like:

`dog.state = "awake" `

I get a negative result:
`{"success":false,"operator":"or","details":{"args":[{"changedTo":["state","awake"],"increased":["hunger"]}],"context":{"state":"awake","hunger":0,"v":{"state":"sleeping","hunger":0},"C":{"hasChangeOperator":true}}},"triggered":false,"hasChangeOperator":true,"stateChange":"maintained-false"}`

And hence the operator of the rule is 'OR'. This should also be successfull. Correct?
